### DISCUSSION TITLE
第3节课作业(18班)

### DISCUSSION THREAD

**Discussion by mattheliu** [InternLM/Tutorial#151] [作业]
请提交作业的链接
Please submit a link to the assignment.

**Comment by yestolife** [comment]
**基础作业**：

复现课程知识库助手搭建过程 (截图)

创建向量数据库：

<img width="750" alt="1" src="https://github.com/InternLM/tutorial/assets/34405348/53192d72-2347-455f-be04-6373e86b954a">
<img width="750" alt="2" src="https://github.com/InternLM/tutorial/assets/34405348/144f5796-e54b-4ab4-88a6-dc0de16af41c">


运行web demo

<img width="960" alt="3" src="https://github.com/InternLM/tutorial/assets/34405348/9afc6458-d4d7-4026-a317-96dc4c1b9a8d">



**进阶作业**：

选择一个垂直领域，收集该领域的专业资料构建专业知识库，并搭建专业问答助手，并在 [[OpenXLab](https://openxlab.org.cn/apps)](https://openxlab.org.cn/apps) 上成功部署（截图，并提供应用地址）

用本课程的md文件制作了一个回答本课程的助手，由于openxlab上的算力未申请下来，只好在InternStudio开发机上跑起来了，能回答课程相关的问题，如InternStudio的显卡是什么，基本回答正确，就是格式还有些问题，见下图。

<img width="960" alt="4" src="https://github.com/InternLM/tutorial/assets/34405348/34d5a509-8d4c-413a-9e68-5555981c2b15">

**Comment by ZMcc18** [comment]
https://www.yuque.com/zixul/avxnzn/yrrq33mzynvoiwrt?singleDoc# 《lesson3 基于 InternLM 和 LangChain 搭建知识库  作业》

**Comment by Heitao5200** [comment]
https://github.com/Heitao5200/my_blog/blob/master/docs/%E7%BB%84%E9%98%9F%E5%AD%A6%E4%B9%A0/%E5%A4%A7%E6%A8%A1%E5%9E%8B%E5%AE%9E%E6%88%98/TASK3.md

**Comment by scorpionfree98** [comment]
[书生 浦语大模型实战营-笔记+作业3](http://t.csdnimg.cn/dtwUm)

**Comment by ggfengxuan** [comment]
[Ⅲ.基于 InternLM 和 LangChain 搭建你的知识库作业](https://v1y50l9n58a.feishu.cn/wiki/Xs0bwbyBTiAIDpkavdlcYwctnad?from=from_copylink)

**Comment by DrZhang1208** [comment]
[书生·浦语第三次作业](https://blog.csdn.net/qq_42866033/article/details/135511467?spm=1001.2014.3001.5501)

**Comment by myaijarvis** [comment]
【参考：[【笔记】书生·浦语大模型实战营——第三课（基于 InternLM 和 LangChain 搭建你的知识库）-CSDN博客](https://blog.csdn.net/Jruo911/article/details/135515038 )】

**Comment by 2001wjh** [comment]
https://www.yuque.com/wjh2001/rdvfen/pdls73wr4a02ckf9?singleDoc# 《基于InternLM和LangChain搭建你的知识库》

**Comment by tackhwa** [comment]
[第3节课笔记+作业](https://nutritious-chill-7a9.notion.site/InternLM-LangChain-5fe5a09f8e3b4ea38d9dadf3223f5e8d?pvs=4)

**Comment by caohuikang** [comment]
创建向量数据库并运行demo
![image](https://github.com/InternLM/tutorial/assets/22697135/e5cd9717-172f-45b8-bbdc-6812b3dc8130)
问答界面
![image](https://github.com/InternLM/tutorial/assets/22697135/546d4f85-fba8-4029-a4f4-aa307e0fc737)

**Comment by xwhclaire** [comment]
[Assignment 3 基于InternLM和LangChain搭建个人知识库](https://github.com/xwhclaire/StudyPackages/blob/main/LLM%20training/Assignment%203%20%E5%9F%BA%E4%BA%8EInternLM%E5%92%8CLangChain%E6%90%AD%E5%BB%BA%E7%9A%84%E4%B8%AA%E4%BA%BA.md)

**Comment by ZiHAO-LI-cmd** [comment]
[书生·浦语大模型实战营 Lesson 3](http://t.csdnimg.cn/LdRNr)

**Comment by caohuikang** [comment]
XTuner 单卡低成本微调你的大模型
![xtuner](https://github.com/InternLM/tutorial/assets/22697135/49ea966a-a4cf-46c7-b7ae-94b0e5860068)

**Comment by mattheliu** [comment]
https://github.com/mattheliu/gitblog/issues/3
